helmrelease/condition_set.go

```diff
@@ -0,0 +1,184 @@
+package helmrelease
+
+import (
+	"context"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2esetup/k8s"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type conditionSetConfig struct {
+	K8sClients *k8s.Clients
```

Comment:
Our k8s client collection is used here, which also has the ability to provide a rest config.